### PR TITLE
decorate classes

### PR DESCRIPTION
**Pull request by ifzyy**
### In this milestone

- i created a class to capitalize lowercase name
- I created a  class to check that name does not go pass 10 characters

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (MahmoudBakr23) [inline decorator.rb:4]: **Required**
- Please add the correct getters and setters for the nameable class as attr accessor :nameable

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @ifzyy 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great work on creating the nameable and decorators classes
- Well done with the custom methods
- Great job specifying the associations between the classes
- Descriptive pr 
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

n/a

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (MahmoudBakr23) [reply decorator.rb:4]: Hi @ifzyy 
Thank you for contacting me back 💯 

In the Decorator class, please add the association `attr_accessor :nameable` in order to connect the two classes with each other. 🙏

Comment (ifzyy) [reply decorator.rb:6]: @MahmoudBakr23  Thanks so much

**Review by DammyShittu — APPROVED** [review]
Hi @ifzyy,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**